aiida_quantumespresso/parsers/pp.py

```diff
@@ -140,12 +124,25 @@ def get_key_from_filename(filename):
             matches = re.search(pattern, filename)
             return matches.group(1)
 
-        for filename, data in data_raw:
-            try:
-                key = get_key_from_filename(filename)
-                data_parsed.append((key, parsers[iflag](data)))
-            except Exception:  # pylint: disable=broad-except
-                return self.exit_codes.ERROR_OUTPUT_DATAFILE_PARSE.format(filename=filename)
+        for filename in filenames:
```

Comment:
Would be good to add a comment here to explain that with this setup, the `data_raw` is parsed directly and then overwritten in the next iteration, which prevents all raw data from being read into memory. Took me a while to realize this from just reading the code

Reply:
Good point, will add a comment!